Yosemite/Yosemite/Actions/ProductAction.swift

```diff
@@ -229,4 +229,28 @@ public enum ProductAction: Action {
                              orderBy: ProductsRemote.OrderKey,
                              order: ProductsRemote.Order,
                              completion: (Result<[ProductReport], Error>) -> Void)
+
+    /// Fetches variation reports for the given product IDs and variation IDs in a given time range.
+    /// - Parameter siteID: Site ID to fetch stock for.
+    /// - Parameter productIDs: IDs of the products of interest.
+    /// - Parameter timeZone: The timezone to consider for the given time range.
+    /// - Parameter earliestDateToInclude: The earliest date for the product reports.
+    /// - Parameter latestDateToInclude: The latest date for the product reports.
+    /// - Parameter pageNumber: The index of the content page to query.
+    /// - Parameter pageSize: The max amount of items to return per page.
+    /// - Parameter orderBy: The key to sort returned items.
+    /// - Parameter order: Whether to sort returned items ASC or DESC.
+    /// - Parameter completion: Closure to trigger when fetching completes.
+    ///
+    case fetchVariationReports(siteID: Int64,
+                               productIDs: [Int64],
+                               variationIDs: [Int64],
```

Comment:
[Here](https://github.com/woocommerce/woocommerce-ios/pull/12803#discussion_r1615451213), I left a question about how the endpoint differentiates different variations when there is a conflict in variation ID value between different products. Please correct me if I am wrong.

Reply:
From the discussions in `p1716794405419699-slack-C03L1NF1EA3`, I believe the variation IDs are unique so we can group them in a single request.